File: Lab7Calificado/Lab7_2026_1/PokeSoftSolution/PokeSoftModel/Pokemon.cs
namespace PokeSoftModel
{
    public class Pokemon
    {
        private int _idPokemon;
        private TipoPokemon? _tipoPokemon;
        private String? _nombre;
        private double _altura;
        private double _peso;
        private EstadoEvolutivo _estadoEvolutivo;
        private String? _descripcion;

        public Pokemon() { }
        public Pokemon(TipoPokemon? tipoPokemon, string? nombre, double altura, double peso, EstadoEvolutivo estadoEvolutivo, string? descripcion)
        {
            _tipoPokemon = tipoPokemon;
            _nombre = nombre;
            _altura = altura;
            _peso = peso;
            _estadoEvolutivo = estadoEvolutivo;
            _descripcion = descripcion;
        }
        public Pokemon(int idPokemon, TipoPokemon? tipoPokemon, string? nombre, double altura, double peso, EstadoEvolutivo estadoEvolutivo, string? descripcion)
        {
            _idPokemon = idPokemon;
            _tipoPokemon = tipoPokemon;
            _nombre = nombre;
            _altura = altura;
            _peso = peso;
            _estadoEvolutivo = estadoEvolutivo;
            _descripcion = descripcion;
        }

        public int IdPokemon { get => _idPokemon; set => _idPokemon = value; }
        public TipoPokemon? TipoPokemon { get => _tipoPokemon; set => _tipoPokemon = value; }
        public String? Nombre { get => _nombre; set => _nombre = value; }
        public double Altura { get => _altura; set => _altura = value; }
        public double Peso { get => _peso; set => _peso = value; }
        public EstadoEvolutivo EstadoEvolutivo { get => _estadoEvolutivo; set => _estadoEvolutivo = value; }
        public String? Descripcion { get => _descripcion; set => _descripcion = value; }
    }
}

File: lab11Calificado/Planchazo-Lab-11-Pokemon/PokemonsProg/PokemonesModel/src/main/java/pe/edu/pucp/pokemones/model/Pokemon.java
package pe.edu.pucp.pokemones.model;

import pe.edu.pucp.pokemones.model.enums.EstadoEvolutivo;

public class Pokemon {
    private int id_Pokemon;
    private int id_tipo_Pokemon;
    private String string_tipo_Pokemon;
    private String nombre;
    private double altura;
    private double peso;
    private EstadoEvolutivo estado_evolutivo;
    private String descripcion;

    public Pokemon(){

    }

    //Getters and setters
    public int getIdPokemon() {
        return id_Pokemon;
    }

    public void setIdPokemon(int id_Pokemon) {
        this.id_Pokemon = id_Pokemon;
    }

    public int getIdTipoPokemon() {
        return id_tipo_Pokemon;
    }

    public void setIdTipoPokemon(int id_tipo_Pokemon) {
        this.id_tipo_Pokemon = id_tipo_Pokemon;
    }

    public String getStringTipoPokemon() {
        return string_tipo_Pokemon;
    }

    public void setStringTipoPokemon(String string_tipo_Pokemon) {
        this.string_tipo_Pokemon = string_tipo_Pokemon;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getAltura() {
        return altura;
    }

    public void setAltura(double altura) {
        this.altura = altura;
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public EstadoEvolutivo getEstadoEvolutivo() {
        return estado_evolutivo;
    }

    public void setEstadoEvolutivo(EstadoEvolutivo estado_evolutivo) {
        this.estado_evolutivo = estado_evolutivo;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }



}
